### ISSUE TITLE
canvas widget

### ISSUE THREAD

**Issue by Akuli** [Akuli/pythotk#5]
as usual, instead of passing around item IDs there should be CanvasItem objects or something

**Comment by PurpleMyst** [comment]
development for this has begun at https://github.com/PurpleMyst/teek/tree/feature/canvas

**Comment by Akuli** [comment]
I have worked on this too https://github.com/Akuli/teek/tree/canvas

**Comment by PurpleMyst** [comment]
That looks pretty complete. Is treeview done too?

**Comment by PurpleMyst** [comment]
Also, I'd appreciate if you could "prune" all the branches which have been merged back into master :)

You can do this by running
```
git branch -r --merged | grep origin | grep -v HEAD | xargs -L1 | sed 's_origin/__' | xargs git push origin --delete --dry-run
```
Without `--dry-run`

**Comment by Akuli** [comment]
I have done nothing to treeview. Would be nice to get that done too.

What does the pruning command do?

**Comment by PurpleMyst** [comment]
It removes all branches that have been merged back in the master and therefore are just noise in the branch list.

I'll see what I can do about treeview later.

**Comment by Akuli** [comment]
tl;dr: canvas is done
